delta/plugins/composite-views/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/compositeviews/stream/CompositeGraphStream.scala

```diff
@@ -62,13 +65,20 @@ object CompositeGraphStream {
       }
     }
 
-    override def rebuild(source: CompositeViewSource, project: ProjectRef): Source = {
+    override def rebuild(
+        source: CompositeViewSource,
+        project: ProjectRef,
+        projectionTypes: Set[Iri]
+    ): Source = {
       source match {
         case p: ProjectSource       =>
-          Source(local.currents(project, p.selectFilter, _).through(drainSource))
+          val filter = p.selectFilter.copy(types = p.selectFilter.types ++ projectionTypes)
```

Comment:
Do we need the source types here at all? Can we just use the projection types?

Reply:
I think it is slightly better to include them in the case where there is a limitation on the source but not in the projection.
In that case, you would have resources that won't be in the common space anyway.
It was also the original behaviour.